File: 2706.购买两块巧克力.ts
/*
 * @lc app=leetcode.cn id=2706 lang=typescript
 *
 * [2706] 购买两块巧克力
 */

// @lc code=start
function buyChoco(prices: number[], money: number): number {
    if (prices.length < 2) return money;

    let [min1, min2] = [+Infinity, +Infinity];
    for (const price of prices) {
        if (price < min1) {
            [min1, min2] = [price, min1];
        } else if (price < min2) {
            min2 = price;
        }
    }

    const remain = money - min1 - min2;

    return remain < 0 ? money : remain;
};
// @lc code=end

File: 2706.购买两块巧克力.rs
/*
 * @lc app=leetcode.cn id=2706 lang=rust
 *
 * [2706] 购买两块巧克力
 */

// @lc code=start
impl Solution {
    pub fn buy_choco(prices: Vec<i32>, money: i32) -> i32 {
        if prices.len() < 2 {
            return money;
        }

        let (mut min1, mut min2) = (i32::MAX, i32::MAX);
        for &p in prices.iter() {
            if p < min1 {
                (min1, min2) = (p, min1);
            } else if p < min2 {
                min2 = p;
            }
        }
        let remain = money - min1 - min2;
        if remain < 0 {
            money
        } else {
            remain
        }
    }
}
// @lc code=end
